### ISSUE TITLE
Components and insertion points

### ISSUE THREAD

**Issue by unsafecode** [knockout/knockout#1592]
Speaking of nesting components, I've reworked a bit the latest available source for the component binding to also support the so called *insertion points*: imagine we want a **bootstrap-panel** component that wraps its content into a bootstrap panel. We would like to write our component as 

```html
<bootstrap-panel>
	<header>
		My panel
	</header>
	<main>
		<strong>My content</strong>
	</main>
	<footer>
		My footer
	</footer>
</bootstrap-panel>
```

Then, our template could look like:

```html
<div class="panel panel-default">
    <div class="panel-heading">
        <content select="header" />
    </div>
    <div class="panel-body">
        <content select="main" />
    </div>
    <div class="panel-footer">
       <content select="footer" />
    </div>
</div>
```

The key feature here is the `<content select="..." />` element, which specifies an *insertion point* by means of a *querySelectorAll* expression. At runtime, before binding occurs, and when the component template is loaded, I've set a function that processes all insertion points, looking a the newly available *originalChildNodes*.

The final output, then, will be:

```html
<bootstrap-panel>
	<div class="panel panel-default">
		<div class="panel-heading">
			<header>
				My panel
			</header>
		</div>
		<div class="panel-body">
			<main>
				<strong>My content</strong>
			</main>
		</div>
		<div class="panel-footer">
			<footer>
				My footer
			</footer>
		</div>
	</div>
</bootstrap-panel>
```

I really hope this approach can prove to be useful. At present time, though, I needed to edit the component binding source, since there's no pluggable way the edit it easliy, so if any of you are interested, I can probably build a PR and make it available.

**Comment by VoiceOfWisdom** [comment]
I modified @vamp's interpolation implementation to use the parent binding context, and support durandal style `data-part` selectors.

[Here.](http://jsbin.com/tojahahefo/1/edit?html,js,output)

**Comment by VoiceOfWisdom** [comment]
[New version that uses a binding handler instead of generating a massive with.](http://jsbin.com/boxabebiqu/1/edit?js,output)

**Comment by novascreen** [comment]
i played around with that jsfiddle a bit more and also added the ability to use `<content></content>` for the remaining nodes
http://jsfiddle.net/q4ydg5mz/2/
i'm wondering, should this line `$content = $templateNodes.filter(query);` be using `filter` or rather do a deep `find` to be closer to what shadow dom does?

**Comment by SteveSanderson** [comment]
I *think* this discussion has moved over to #1665, and may even be resolved with the `componentTemplateNodes` feature we shipped in KO 3.3.0.

But if I'm mistaken and this discussion should continue separately here, please reopen and clarify. Thanks!
